### PR TITLE
Catch and handle errors thrown by plugin main fn. Fixes #201

### LINKED ISSUE #201: Errors thrown by a plugin's `main` function should be handled automatically
Async errors thrown by a plugin's `main` function currently fail silently. These should be handled correctly without requiring manual `try/catch` blocks or `.catch` on promises. I believe this is also an issue with sync errors but I would have to confirm this...

### PR DESCRIPTION
**Pull request by brollb**

### CODE REVIEW

**Comment by brollb** [comment]
@kecso - It might also be nice to have the plugin set the success flag on the result... Is there any reason why it doesn't do that already? Is there any case where the plugin execution will not return an error but also not be successful?

**Comment by kecso** [comment]
> @kecso - It might also be nice to have the plugin set the success flag on the result... Is there any reason why it doesn't do that already? Is there any case where the plugin execution will not throw an error but also not be successful?

if you meant throw error, then it is different. success means that it successfully executed its task and got what it wanted, if that is not the case (some data missing, the model is not in its right state, etc...), the plugin should still finish gracefully without throwing an error, but it should send a plugin-error about the situation. When it throws, that show a deeper problem, something the plugin was not expecting at all...

File: src/plugin/managerbase.js [hunk 1]
```diff
@@ -312,12 +306,38 @@ define([
                     plugin.createMessage(null, 'The main callback is being called more than once.');
                     plugin.createMessage(null, stackTrace);
                     callback('The main callback is being called more than once!', result);
+                }
+            };
+            try {
+                if (expectsCallback) {
+                    result = await new Promise((resolve, reject) =>
```
Comment (kecso) [inline src/plugin/managerbase.js:313]: is Promise supported by all browsers?

Comment (brollb) [reply src/plugin/managerbase.js:313]: Good question. How far back do you care to support? https://caniuse.com/#search=promise

**Pull request merged**